chainntnfs/btcdnotify/btcd_debug.go

```diff
@@ -0,0 +1,80 @@
+// +build debug
+
+package btcdnotify
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/lightningnetwork/lnd/chainntnfs"
+)
+
+// UnsafeStart starts the notifier with a specified best block.
+// Both its bestBlock and its txConfNotifier are initialized with bestBlock.
+// Used for testing.
+func (b *BtcdNotifier) UnsafeStart(bestBlock chainntnfs.BlockEpoch,
+	syncHeight int32) error {
+
+	// Connect to btcd, and register for notifications on connected, and
+	// disconnected blocks.
+	if err := b.chainConn.Connect(20); err != nil {
+		return err
+	}
+	if err := b.chainConn.NotifyBlocks(); err != nil {
+		return err
+	}
+
+	b.txConfNotifier = chainntnfs.NewTxConfNotifier(
+		uint32(bestBlock.Height), reorgSafetyLimit)
+
+	b.chainUpdates.Start()
+	b.txUpdates.Start()
+
+	// Ensure no block notifications are pending when we start the
+	// notification dispatcher goroutine.
```

Comment:
Would also add that this is needed due to the latency between the miner generating blocks and the notifier actually being notified about them.

Reply:
fixed